src/components/ConfigValidation/ConfigIndicator.tsx

```diff
@@ -53,7 +53,7 @@ const tooltipListStyle = style({
 });
 
 export class ConfigIndicator extends React.PureComponent<Props, {}> {
-  numberOfChecks = (type: string) => (this.props.validation.checks || []).filter(i => i.severity === type).length;
+  numberOfChecks = (type: string) => this.props.validation.checks.filter(i => i.severity === type).length;
```

Comment:
I see the change.
Ensuring that `this.props.validation.checks` makes code cleaner, but I understand that the former code was correct.
I mean the `(this.props.validation.checks || [])` should also protect against nulls.

Or there was some case not well covered.

P.S.: I'm asking more to learn/understand it, sometimes javascript has some "false friends" and we finish doing some things odd.

Reply:
@lucasponce this null-check was ok. The one that wasn't ok was in file `ServiceInfo.tsx` because we were checking against `undefined` while the no-value case was `null`. My change is to say now: ok , the contract is now different, `check` is just non-nullable and non-undefinedable and it is (kinda) guaranteed by the backend.

Reply:
PS: this is all guided by the change in the model where `checks?: ObjectCheck[];` now becomes `checks: ObjectCheck[];`.

Reply:
+1
I like this approach better.
Everywhere where types are fully defined is a win.